### ISSUE TITLE
Elm 19 only has a single `elm` command

### ISSUE THREAD

**Issue by 7sharp9** [Krzysztof-Cieslak/vscode-elm#242]
```
# 0.19         # 0.18
elm make       # elm-make
elm repl       # elm-repl
elm reactor    # elm-reactor
elm install    # elm-package install
elm publish    # elm-package publish
elm bump       # elm-package bump
elm diff       # elm-package diff
```

**Comment by 7sharp9** [comment]
So unless you alias `elm-make` to `elm make` thats the first thing that breaks.

**Comment by 7sharp9** [comment]
Are these just in `package.json`?

**Comment by Punie** [comment]
I believe there is already some discussions in #219 about changes needed for the extension to support 0.19.

**Comment by hakonrossebo** [comment]
If not taken already, I can try to fix and create a pr for this

**Comment by Krzysztof-Cieslak** [comment]
Go ahead @hakonrossebo

**Comment by hakonrossebo** [comment]
Should we try to keep compability with elm 0.18 on this and support both command types based on the Elm version?

**Comment by mainlymortal** [comment]
I realise this isn't a support forum but even after aliasing elm-make to elm make (alias elm-make='elm make') I still get the error "The 'elm-make' compiler is not available."

I also tried change the extension settings but same error. Is this a separate issue on my end or is this related to the current issue?

**Comment by 7sharp9** [comment]
I dont think the alias works

**Comment by 7sharp9** [comment]
I was trying but I cant get the current cloned repo to build?

**Comment by 7sharp9** [comment]
you can modify the source in the addin.

Find `ElmMake.js` modify line 30 to:
```
let args = ['make', file, '--output=' + name];
```
Then ensure the elm-make is updated to the new elm command in your config:
```
"elm.makeCommand": "elm",
```

**Comment by mainlymortal** [comment]
That doesn't appear to fix it (I understand it is just temporary anyway). A new error states  - "There is no FileNameHere.elm command."

**Comment by Punie** [comment]
Is anyone working on a PR for elm 0.19 support then ? I'd be happy to help if it's a work in progress and feedback is needed.

**Comment by 7sharp9** [comment]
@mainlymortal Sounds like you put `'make'` in the wrong place, it needs to go before the file argument.

**Comment by 7sharp9** [comment]
In terms of a PR this would only fix the Elm make command, theres so much else thats completely broke.  A PR for this issue would be to detect the presence of `elm-make` to retain the existing behaviour, then detect just elm to switch to the new behaviour of using `make` as the first parameter.  Oh, also `--warn` is also now not a valid option so that has to be removed from the arguments list too.

**Comment by 7sharp9** [comment]
I would do a PR, but this repo doesn't build, none of the type script references seem to be found so I just get errors on building a fresh repo

**Comment by 7sharp9** [comment]
ok, got it working, had to reinstall yarn, let me see if I can hack something to fix at least `elm make`

**Comment by hakonrossebo** [comment]
@7sharp9 - I've started a PR to fix the commands, and started with some logic to detect the Elm version and fixing the linter - If you want, you can continue on my branch? https://github.com/hakonrossebo/vscode-elm/tree/elm19commands

**Comment by 7sharp9** [comment]
@hakonrossebo Ah, lets try and combine them then, I already started a PR too #243 :-)

**Comment by hakonrossebo** [comment]
If you have time and want to continue - you can copy any code from my commits if any help. I don't have time to do anything before later this evening.

**Comment by 7sharp9** [comment]
@hakonrossebo I can cherry-pick your commit into the PR and use your version detection rather than relying on the elm-make command being set to elm-make for v 0.18, and then we can whittle through the other issues as we get time.  Anyone else can also add to the PR as we go too...

**Comment by hakonrossebo** [comment]
Please do that. I was at the point of investigating the json report output from elm make - Seems to have a different structure.

**Comment by 7sharp9** [comment]
@hakonrossebo Did you mean too add a new section to config `compiler` thats set to `elm`?   I'll add that now if so.

**Comment by hakonrossebo** [comment]
yes - the purpose for having this in config would be to enable users to specify this by full path if needed.

**Comment by 7sharp9** [comment]
I added it, just adjusting the normal make command to use the version detection you put in, then I'll have to leave this until later...

**Comment by Krzysztof-Cieslak** [comment]
Fixed by #243

**Comment by Krzysztof-Cieslak** [comment]
Elm 0.19 integration released in `0.20.0`. Thanks a lot for all the work folks! ❤️
